changelog.d/gh-9628.fixed

```diff
@@ -0,0 +1,3 @@
+When Semgrep fails to produce a bytepos from a line/col position (an internal
+error that should be impossible), we previously silently moved on. This causes
+problems with Autofix, however, so it's better to give a clear error message.
```

Comment:
I usually like to provide an example of something that _now_ works.

Reply:
Hm, this is pretty hard because we're changing how an error state that should be impossible is reported. I'll try another version

Reply:
I think I figured it out!